src/components/DailyQuiz.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState, useEffect } from "react";
+import quizData from "../data/quizData.json";
+
+const DailyQuiz = () => {
+  const [currentDay, setCurrentDay] = useState(0);
+  const [quizDayIndex, setQuizDayIndex] = useState(0);
+  const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
+  const [score, setScore] = useState(0);
+  const [showAnswer, setShowAnswer] = useState(false);
+  const [selectedAnswer, setSelectedAnswer] = useState(null);
+  const [progress, setProgress] = useState(0);
+  const [quizCompleted, setQuizCompleted] = useState(false);
+
+  // Unlocks quizzes based on the system's current date
+  useEffect(() => {
+    const today = new Date().toLocaleDateString(); // Format: MM/DD/YYYY
+    const lastCompletedDate = localStorage.getItem("quizLastCompletedDate");
+
+    if (lastCompletedDate !== today) {
+      const nextDay = currentDay + 1;
+      setCurrentDay(nextDay < quizData.length ? nextDay : currentDay);
+      setQuizDayIndex(nextDay - 1); // Set the active quiz to the latest unlocked day
```

Comment:
should setQuizDayIndex also have the same conditional statement as currentday on the previous line?

Reply:
I fixed it. Thanks for the review!